src/main/java/org/junitpioneer/jupiter/issue/IssueExtensionExecutionListener.java

```diff
@@ -93,23 +94,22 @@ public void testPlanExecutionFinished(TestPlan testPlan) {
 	}
 
 	List<IssueTestSuite> createIssueTestSuites() {
-		//@formatter:off
 		return testCases
-				.values().stream()
-				.collect(toMap(IssueTestCaseBuilder::getIssueId, builder -> new ArrayList<>(List.of(builder)),
-						(builders1, builders2) -> {
-							builders1.addAll(builders2);
-							return builders1;
-						}))
-				.entrySet().stream()
-				.map(issueIdWithTestCases -> new IssueTestSuite(
-						issueIdWithTestCases.getKey(),
-						issueIdWithTestCases
-								.getValue().stream()
-								.map(IssueTestCaseBuilder::build)
-								.collect(toUnmodifiableList())))
+				.values()
+				.stream()
+				.collect(toMap(IssueTestCaseBuilder::getIssueId, this::getIssueTestCases, this::mergeIssueTestCases))
+				.entrySet()
+				.stream()
+				.map(entry -> new IssueTestSuite(entry.getKey(), entry.getValue()))
 				.collect(toUnmodifiableList());
-		//@formatter:on
+	}
+
+	private List<IssueTestCase> getIssueTestCases(IssueTestCaseBuilder t) {
```

Comment:
Why `t`, for "test" / "test case"? If you want to stick to a single character, maybe `b`? (I think I would use `builder`.)